crates/swc_xml_parser/src/lexer/mod.rs

```diff
@@ -0,0 +1,3046 @@
+
+            self.attribute_start_position = Some(self.cur_pos);
+        }
+    }
+
+    fn append_to_attribute(
+        &mut self,
+        name: Option<(char, char)>,
+        value: Option<(bool, Option<char>, Option<char>)>,
+    ) {
+        if let Some(Tag { attributes, .. }) = &mut self.current_tag_token {
+            if let Some(attribute) = attributes.last_mut() {
+                if let Some(name) = name {
+                    attribute.name.push(name.0);
+
+                    if let Some(raw_name) = &mut attribute.raw_name {
+                        raw_name.push(name.1);
+                    }
+                }
+
+                if let Some(value) = value {
+                    if let Some(c) = value.1 {
+                        if let Some(old_value) = &mut attribute.value {
+                            old_value.push(c);
+                        } else {
+                            let mut new_value = String::with_capacity(255);
+
+                            new_value.push(c);
+
+                            attribute.value = Some(new_value);
+                        }
+                    }
+
+                    if let Some(c) = value.2 {
+                        // Quote for attribute was found, so we set empty value by default
+                        if value.0 && attribute.value.is_none() {
+                            attribute.value = Some(String::with_capacity(255));
+                        }
+
+                        if let Some(raw_value) = &mut attribute.raw_value {
+                            raw_value.push(c);
+                        } else {
+                            let mut raw_new_value = String::with_capacity(255);
+
+                            raw_new_value.push(c);
+
+                            attribute.raw_value = Some(raw_new_value);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn update_attribute_span(&mut self) {
+        if let Some(attribute_start_position) = self.attribute_start_position {
+            if let Some(Tag {
+                ref mut attributes, ..
+            }) = self.current_tag_token
+            {
+                if let Some(last) = attributes.last_mut() {
+                    last.span =
+                        Span::new(attribute_start_position, self.cur_pos, Default::default());
+                }
+            }
+        }
+    }
+
+    fn set_tag_to_empty_tag(&mut self) {
+        if let Some(Tag { kind, .. }) = &mut self.current_tag_token {
+            *kind = TagKind::Empty;
+        }
+    }
+
+    fn emit_tag_token(&mut self, kind: Option<TagKind>) {
+        if let Some(mut current_tag_token) = self.current_tag_token.take() {
+            if let Some(kind) = kind {
+                current_tag_token.kind = kind;
+            }
+
+            match current_tag_token.kind {
+                TagKind::Start => {
+                    let mut already_seen: AHashSet<String> = Default::default();
```

Comment:
I think this should be `AHashSet<JsWord>`

Reply:
To avoid cloning memory

Reply:
@kdy1 Fixed :+1: